xtask/src/dependencies.rs

```diff
@@ -0,0 +1,102 @@
+use std::{collections::HashMap, time::Instant};
+
+use crate::{
+    endgroup, group,
+    logging::init_logger,
+    utils::{
+        ensure_cargo_command_is_installed, format_duration, is_current_toolchain_nightly,
+        run_cargo, Params,
+    },
+};
+
+#[derive(clap::ValueEnum, Default, Copy, Clone, PartialEq, Eq)]
+pub(crate) enum DependencyCheckType {
+    /// Run all dependency checks.
+    #[default]
+    All,
+    /// Perform an audit of all dependencies using the cargo-audit crate `<https://crates.io/crates/cargo-audit>`
+    Audit,
+    /// Run cargo-deny check `<https://crates.io/crates/cargo-deny>`
+    Deny,
+    /// Run cargo-udeps to find unused dependencies `<https://crates.io/crates/cargo-udeps>`
+    Unused,
+}
+
+pub(crate) fn run(dependency_check: DependencyCheckType) -> anyhow::Result<()> {
+    // Setup logger
+    init_logger().init();
```

Comment:
You can do:

```rust
impl DependencyCheckType {
    pub(crate) fn run(self) -> anyhow::Result<()> {
        ...
    }
}
```

To allow the dot notation:

```rust
DependencyCheckType::All.run();
```

Very very optional, just if you think it's cleaner.

Reply:
Done